middleware/app/Http/Controllers/Api/V1/PlatformConnectionController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Http\api\V1\Controllers\Controller;
+use App\Http\Requests\PlatformConnectionRequest;
+use App\Http\Requests\ShowPlatformConnectionRequest;
+use App\Http\Requests\ShowStudentEnrollmentRequest;
+use App\Models\PlatformConnection;
+use App\Http\Resources\PlatformConnectionResource;
+
+class PlatformConnectionController extends Controller
+{
+
+    /* Get all platform connections */
+    //*************************************************************
+    // GET: /api/platform_connection/
+    // Request $req example:
+    // { "consumer_id": 1 }
+    // *************************************************************
+    public function index(): PlatformConnectionResource
+    {
+        return PlatformConnectionResource::collection(PlatformConnection::all());
+    }
+
+    /* Create a new platform connection */
+    //*************************************************************
+    // POST: /api/platform_connection/
+    // PlatformConnectionRequest $req example:
+    // { "consumer_id": 1, "provider_id": 1, "state": "enabled" }
+    // *************************************************************
+    public function store(PlatformConnectionRequest $req): PlatformConnectionResource
+    {
+        $validated = $req->validated();
+        $exists = PlatformConnection::where($validated)->first();
+        if ($exists) {
+            return response()->json(['error' => 'Platform connection already exists'], 401);
+        }
+        try {
+            $platform_connection = PlatformConnection::create($req->validated());
+            return new PlatformConnectionResource($platform_connection);
+        } catch (\Exception) {
+            return response()->json(['error' => 'Invalid request body'], 401);
+        }
+    }
+
+    // Get a specific platform connection by consumer or provider id
+    // *************************************************************
+    // GET: /api/platform_connection/{id}
+    // Request $req example:
+    //  "consumer_id": 1 || "provider_id": 1
+    // *************************************************************
+    public function show(ShowPlatformConnectionRequest $req): PlatformConnectionResource
+    {
+        try {
+            $validated = $req->validated();
```

Comment:
I may be mistaken, but according to the validation rules in `ShowPlatformConnectionRequest`, if the consumer_id is required if the provider_id is present and vice/versa. Wouldn't that mean that both would have to be present to pass validation, thus negating the OR functionality?